AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -25,6 +26,41 @@ static void runLoopSourceCallback(void *info) {
 #endif
 }
 
+#pragma mark - ASDeallocThread
+
+@interface ASDeallocThread : NSThread
+@property (nonatomic, strong) ASRunLoopQueue *deallocQueue;
+@end
+
+@implementation ASDeallocThread
+
+- (void)main
+{
+  self.name = @"ASDeallocThread";
+  [self deallocQueue];
+  while (YES) {
+    // This method will still return once all sources have finished.
+    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate distantFuture]];
```

Comment:
I think I figured it out.  CFRunloopRun() is actually, apparently, totally different than ANY option available on NSRunloop.  NSRunloop run will return, but CFRunloopRun will continue "indefinitely" per the documentation.  What a sham :-P